linker/shared/src/main/scala/org/scalajs/linker/checker/ClassDefChecker.scala

```diff
@@ -913,20 +920,54 @@ private final class ClassDefChecker(classDef: ClassDef,
     if (!declaredLabelNamesPerMethod.add(label.name))
       reportError(i"Duplicate label named ${label.name}.")
   }
+
+  private def checkLinkTimeCond(cond: LinkTimeTree)(
+      implicit ctx: ErrorContext): Unit = {
+    if (cond.tpe != BooleanType)
+      reportError(i"Link time condition must be typed as boolean, but ${cond.tpe} is found.")
```

Comment:
I think we can merge this directly into `case LinkTimeIf`. It's the `LinkTimeIf` that needs its `cond` to be a boolean. The condition itself doesn't care.

It's also more similar to how `case If` gets type-checked in the IR checker.